clients/contracts/token.ts

```diff
@@ -31,17 +31,14 @@ export class TokenContract extends DelphinusContract {
 
   mint(amount: number) {
     const pbinder = new PromiseBinder();
+
     return pbinder.return(async () => {
-      return await pbinder.bind(
-        "Mint",
-        this._mint(amount)
-      );
+      await pbinder.bind("Mint", console.log("Minting"));
```

Comment:
Are you trying to add a snapshot?

Reply:
If all you want to do is to add a snapshot here, you can write pbinder.snapshot("Mint"); And lateron you can use XXX.when("snapshot", "Mint" ...) to capture this. There is not use to add two steps "Mint" and "Transfer" here, since mint is a single process transaction, we can add a Mark(Snapshort）before the tx begin by snapshot("Mint") and then bind a process "Mint" later on.